packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1111,6 +1111,19 @@ class PipelineOwner {
 /// The [RenderBox] subclass introduces the opinion that the layout
 /// system uses Cartesian coordinates.
 ///
+/// ## Lifecycle
+///
+/// A [RenderObject] must [dispose] when it is no longer needed. The creator
+/// of the object is responsible for disposing of it. Typically, the creator is
+/// a [RenderObjectElement], and that element will dispose the object it creates
+/// when it is unmounted.
+///
+/// [RenderObject]s are responsible for cleaning up any expensive resources
+/// they hold when [dispose] is called, such as [Picture] or [Image] objects.
+/// This includes any [Layer]s that the render object has directly created. By
+/// default, the base implementation of dispose will dispose of the [layer], and
```

Comment:
I don't think subclasses can change this "default" behavior since they have to call `super.dispose`? Should we maybe just say that Layers created by the render object must be disposes manually unless they have been set to the layer property, which is automatically disposed?

Reply:
Also: This patch doesn't contain any layer disposing mechanics (it just nulls out the property). Do you plan to add that logic to this PR or will that be added later? If it is latter, maybe we should just document what's implemented for now?

Reply:
Reworded this a bit -layer.dispose will be in a separate patch, instead just talking about nulling the property.